Userland/Services/LaunchServer/Launcher.cpp

```diff
@@ -312,6 +312,19 @@ bool Launcher::open_file_url(const URL& url)
     String extension = {};
     if (extension_parts.size() > 1)
         extension = extension_parts.last();
-    return open_with_user_preferences(m_file_handlers, extension, url.path(), "/bin/TextEditor");
+
+    // Additionnal parameters parsing, specific for the file protocol
+    Vector<String> additional_parameters;
+    additional_parameters.append(url.path());
+    auto parameters = url.query().split('&');
+    for (auto parameter = parameters.begin(); parameter != parameters.end(); ++parameter) {
+        auto pair = parameter->split('=');
+        if (pair[0] == "line_number") {
+            auto line = pair[1].to_int();
+            if (line.has_value())
+                additional_parameters.prepend(String::format("-l%i", *line));
```

Comment:
Please use `String::formatted()` in new code, also I think `value()` is more clear than `operator*()` here

```suggestion
                additional_parameters.prepend(String::formatted("-l {}", line.value()));
```